test/query_test.js

```diff
@@ -78,19 +78,17 @@ describe('query', () => {
   })
 
   it('map', async function() {
-    const double = query.lambda(x => query.multiply([2, x]))
-    await assertQuery(query.map(double, [1, 2, 3]), [2, 4, 6])
+    await assertQuery(query.map(a => query.multiply([2, a]), [1, 2, 3]), [2, 4, 6])
 
-    const getN = query.lambda(x => query.select(['data', 'n'], query.get(x)))
     const page = query.paginate(nSet(1))
-    const ns = query.map(getN, page)
+    const ns = query.map(a => query.select(['data', 'n'], query.get(a)), page)
     assert.deepEqual((await client.query(ns)).data, [1, 1])
   })
 
   it('foreach', async function() {
     const refs = [(await create()).ref, (await create()).ref]
     await client.query(
-      query.foreach(query.lambda(query.delete_expr), refs))
```

Comment:
since this still technically supports passing in a lambda construct as opposed to a function, (query.lambda, etc) we should probably have a test for that still too. unless there is one somewhere outside of the diff and im missing it